examples/stable-diffusion/training/README.md

```diff
@@ -0,0 +1,207 @@
+  model_id,
+  torch_dtype=torch.bfloat16,
+  use_habana=True,
+  use_hpu_graphs=True,
+  gaudi_config="Habana/stable-diffusion",
+)
+
+prompt = "A <cat-toy> backpack"
+
+image = pipe(prompt, num_inference_steps=50, guidance_scale=7.5).images[0]
+
+image.save("cat-backpack.png")
+```
+
+
+## Fine-Tuning
+
+The `train_text_to_image.py` script shows how to implement the fine-tuning of Stable Diffusion models on Habana Gaudi.
+
+### Example for SDXL
+We can launch the fine-tuning of SDXL model using:
+
+```bash
+python train_text_to_image.py \
+  --pretrained_model_name_or_path stabilityai/stable-diffusion-xl-base-1.0 \
+  --pretrained_vae_model_name_or_path stabilityai/sdxl-vae \
+  --dataset_name lambdalabs/pokemon-blip-captions \
+  --resolution 1024 \
+  --center_crop \
+  --random_flip \
+  --proportion_empty_prompts=0.2 \
+  --train_batch_size 1 \
+  --gradient_accumulation_steps 4 \
+  --max_train_steps 3000 \
+  --learning_rate 1e-05 \
+  --max_grad_norm 1 \
+  --lr_scheduler constant \
+  --lr_warmup_steps 0 \
+  --output_dir sdxl-pokemon-model \
+  --gaudi_config_name Habana/stable-diffusion \
+  --throughput_warmup_steps 3 \
+  --use_hpu_graphs \
+  --bf16
+```
+
+### Example for LoRA SDXL
+
+Low-Rank Adaption (LoRA) allows adapting a pretrained model by adding pairs of rank-decomposition matrices to
+existing weights and only training those newly added weights.
+
+We can launch the LoRA based fine-tuning of SDXL model using:
+
+```bash
+python train_text_to_image.py \
+  --pretrained_model_name_or_path="stabilityai/stable-diffusion-xl-base-1.0" \
+  --pretrained_vae_model_name_or_path="madebyollin/sdxl-vae-fp16-fix" \
+  --dataset_name="lambdalabs/pokemon-blip-captions" \
+  --caption_column="text" \
+  --resolution=1024 --random_flip \
+  --train_batch_size=1 \
+  --num_train_epochs=2 --checkpointing_steps=500 \
+  --learning_rate=1e-04 --lr_scheduler="constant" --lr_warmup_steps=0 \
+  --seed=42 \
+  --output_dir="sd-pokemon-model-lora-sdxl" \
+  --finetuning_method="lora" \
+  --gaudi_config_name="Habana/stable-diffusion" \
+  --throughput_warmup_steps=3 \
+  --use_hpu_graphs \
+  --bf16
+```
+
+**Notes:**
+
+* SDXL's VAE is known to suffer from numerical instability issues. This is why we
+also expose a CLI argument namely `--pretrained_vae_model_name_or_path` that lets
+you specify the location of a better VAE (such as this one).
+
+#### LoRA SDXL Inference
+
+Once you have trained a LoRA weights as in the example above, inference can be done
+by using the `GaudiStableDiffusionXLPipeline`.
+
+```python
+import torch
+from optimum.habana.diffusers import (
+    GaudiStableDiffusionXLPipeline,
+    GaudiEulerDiscreteScheduler,
+)
+
+model_id = "stabilityai/stable-diffusion-xl-base-1.0"
+lora_model_id = "sd-pokemon-model-lora-sdxl"
+pipe = GaudiStableDiffusionXLPipeline.from_pretrained(
+  model_id,
+  scheduler=GaudiEulerDiscreteScheduler.from_pretrained(model_id, subfolder="scheduler"),
+  torch_dtype=torch.bfloat16,
```

Comment:
can you change GaudiEulerDiscreteScheduler to DDPMScheduler like original script as we observed noise image with EulerDiscreteScheduler ?

Reply:
For this change, do we need to implement GaudiDDDPMScheduler?

Reply:
I guess yes. Maybe it works well with `GaudiDDIMScheduler`?

Reply:
@regisss  @libinta  we tested now with updated diffusers Euler Discrete scheduler works good (no noise images any more observed)